highfidelity/hifi
# Snapshots: Fix incorrect "for_url share

Two fixes here: One to prevent the incorrect multiple-uploads of previously-taken snapshots, and the other to prevent the Snapshot script from incorrectly sharing old Snapshots in certain cases.

This fixes [FB5017](https://highfidelity.fogbugz.com/f/cases/5017/Previously-taken-snapshots-are-getting-uploaded-to-the-server-with-audience-for_url-when-they-shouldn-t) and [FB5019](https://highfidelity.fogbugz.com/f/cases/5019/When-clicking-Share-on-a-Snapshot-sometimes-an-old-Snapshot-gets-shared).

**Test Plan:**
Clear your settings. You can use desktop or HMD mode for this test plan.
1. Open the Snapshot app. Take a snapshot. Share it to the snaps feed.
2. Open the GoTo app and verify that the snapshot from (1) is visible in the feed.
3. Open the Snapshot app. Take a NEW snapshot. Share it to the snaps feed.
4. Open the GoTo app and verify that BOTH the snapshot from (1) AND the snapshot from (3) are visible in the feed.
5. Open your logs with CTRL+SHIFT+L
6. Take a still snapshot using the Snapshot app
7. Notice from the logs that the still snapshot (correctly) gets uploaded with audience "for_url"
8. Close the Snapshot app
9. Open the Snapshot app
10. Verify that you don't see any logs about uploading the still snapshot with audience "for_url"

To verify that FB5019 is fixed, you'll have to be _fast_! Here's the test plan for that:

1. Using the Snapshot app, take a Still-and-GIF snapshot
2. Move your viewpoint 180 degrees.
3. IMMEDIATELY after the red capture button is re-enabled, click it to take another snapshot
4. IMMEDIATELY after the GIF finishes processing, click "Share to Snaps Feed"
5. Wait until you see confirmation that the snapshot was shared
6. Check the Snaps feed, and verify that the GIF Snapshot from step (3) (the second snapshot) is what's visible in the feed
